### DISCUSSION TITLE
List of supported Components

### DISCUSSION THREAD

**Discussion by TheHadiAhmadi** [ubuilder/cms#7] [General]
- [x] Autocomplete
- [x] Accordion
- [x] Avatar
- [x] Alert
- [x] Button
- [x] Badge
- [x] Breadcrumb
- [x] ButtonList/ButtonGroup
- [ ] ColorPicker
- [x] Checkbox
- [x] Card
- [x] Dropdown
- [ ] DatePicker
- [x] Form
- [ ] FileUpload
- [x] Icon
- [x] Input
- [x] Image
- [x] Modal
- [ ] Offcanvas
- [ ] Navbar
- [x] Radio
- [x] RadioGroup
- [x] Select
- [x] Textarea
- [ ] TextEditor
- [x] Spinner
- [x] Switch
- [x] Progress
- [x] Divider
- [x] Tooltip
- [x] Tab
- [x] Table
- [x] Popover
- [ ] Fieldset
- [x] View

**Comment by TheHadiAhmadi** [comment]
AutoComplete features:

- [ ] items prop
- [ ] value (array or any)
- [ ] multiple (boolean)
- [ ] label
- [ ] placeholder
- [ ] disabled
- [ ] create (+ onCreated event)
- [ ] load data from api
- [ ] onSelect event
- [ ] not found / loading slots
- [ ] ....

(check Tom Select)

**Comment by TheHadiAhmadi** [comment]
Alert features:

- [x] color / state / type / variant ..
- [x] icon
- [x] description
- [x] title
- [ ] ...

**Comment by TheHadiAhmadi** [comment]
Avatar features:

- [x] size
- [ ] shape (only circle for now)
- [x] image url
- [x] initial text
- [ ] ...

**Comment by TheHadiAhmadi** [comment]
Button features:

- [x] href (if href -> tag should be `a`)
- [x] size (sm, md, lg)
- [x] color (primary, secondary, success, warning, info, error, light, dark)
- [x] link (boolean)
- [ ] outline (boolean)
- [x] onClick event
- [ ] ...

**Comment by TheHadiAhmadi** [comment]
Badge features

- [x] color
- [x] size
- [ ] ...

**Comment by TheHadiAhmadi** [comment]
Breadcrumb features

- [x] Breadcrumb and Breadcrumb.Item components
- [x] item have href prop
- [ ] ...

**Comment by TheHadiAhmadi** [comment]
ButtonGroup features:

- [x] compact (boolean)
- [ ] ...

**Comment by TheHadiAhmadi** [comment]
all Form components should be inside `Form` component

```js
function SignupForm ({slot, ...restProps}) {
    return Form({
        slot: [
            Form.Input({name: 'username', label: 'Username'}),
            Form.Select({name: 'gender', label: 'Gender'}),
            Form.Checkboxes({
                name: 'languages', 
                label: 'Spoken Languages', 
                items: ["FA", "PS", "EN"]
            }),
            ...slot
        ],
        ...restProps
    })
}

```

**Comment by TheHadiAhmadi** [comment]
@JawadAzizi 
Which form example is better? (above is easier to implement and below will be easier to use)
above form or this?

is it possible to support below form?
```js
import { Form } from '@ulibs/ui'

export function SignupForm ({slot, ...restProps}) {
    return Form({
        fields: {
            username: Form.Input({ label: 'Username' }),
            gender: Form.Select({ label: 'Gender' }),
            languages: Form.Checkboxes({
                label: 'Spoken Languages', 
                items: ["FA", "PS", "EN"]
            })
        },
        onSubmit: (data) {
            // this will run in backend...
            const user = await UserModel.insert(data)
            return {
                redirect: '/users/edit?id=' + user.id
            }
        },
        ...restProps
    })
}

```

**Reply by JawadAzizi** [reply]
I prefer the above one

**Comment by TheHadiAhmadi** [comment]
Use CahtGPT to do boring tasks (writing docs and examples, initial code of components, ....)

**Comment by TheHadiAhmadi** [comment]
Fieldset Features:
Should be similar to https://www.yesvelte.com/docs/form/form-fieldset
- [ ] Legend

**Comment by TheHadiAhmadi** [comment]
Progress Features:

- [ ] color
- [ ] size
- [ ] value
- [ ] indeterminate
- [ ] label (not important)

**Comment by TheHadiAhmadi** [comment]
Divider Features

- [ ] Text (and without text)
- [ ] placement (start, center, end)
- [ ] direction (horizontal & vertical)

**Comment by TheHadiAhmadi** [comment]
Image features:
(just a wapper around img tag)

- [ ] src
- [ ] .... (add other features..)

**Comment by TheHadiAhmadi** [comment]
Spinner features:

- [ ] size
- [ ] color
